Sinetek-rtsx/3rdParty/linux/drivers/misc/cardreader/rts5249.c

```diff
@@ -28,14 +28,14 @@ int rtsx_write_phy(struct rtsx_softc *, u_int8_t, u_int16_t);
 static void rts5249_fill_driving(struct rtsx_pcr *pcr, u8 voltage)
 {
 	u8 driving_3v3[4][3] = {
-		{0x11, 0x11, 0x18},
-		{0x55, 0x55, 0x5C},
-		{0xFF, 0xFF, 0xFF},
+		{0x13, 0x13, 0x13},
```

Comment:
> I would suggest adding a new file (e.g: rts5227.c) with its own methods instead of modifying this one.

It doesn't seem to make sense to change here.
I have restored it before merging. What I did was add IOSleep() to Sinetek_rtsx::setPowerState.

Reply:
I agree with @CLAY-BIOS (I understand those changes are for rts5227, and they probably will affect 5249 and 525A). If you really need those, I think you should add a new file, rts5227.c, based on [this one](https://github.com/torvalds/linux/blob/master/drivers/misc/cardreader/rts5227.c), and use it's functions when the chip is detected to actually be rts5227. Otherwise maintenance is unmanageable.